### PR TITLE
BZ1894063: Added important block on using pods deployment in SSC

### PR DESCRIPTION
**Pull request by xenolinux**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1894063

OCP version: 4.8+

Preview: https://53264--docspreview.netlify.app/openshift-enterprise/latest/authentication/managing-security-context-constraints.html#security-context-constraints-about_configuring-internal-oauth

@stlaz @cardil PTAL and let me know your feedback.

@wangke19 PTAL for QA review.

QE review: lgtm

### CODE REVIEW

File: modules/security-context-constraints-about.adoc [hunk 1]
```diff
@@ -8,6 +8,11 @@
 
 Similar to the way that RBAC resources control user access, administrators can use _security context constraints_ (SCCs) to control permissions for pods. These permissions include actions that a pod can perform and what resources it can access. You can use SCCs to define a set of conditions that a pod must run with to be accepted into the system.
 
+[IMPORTANT]
+====
+There is a difference in permissions experienced by containers, when you create pods directly by `kubectl` or `oc` and when pods are being created by some higher level type like deployment or job. The pods created by higher level resources use secured `ServiceAccount` access, but creating the pod uses `kubectl` or `oc` user access, which may exceed safe {product-title} standards. The set of SCCs that admission uses to authorize a pod are determined by the user identity and groups that the user belongs to. Additionally, if the pod specifies a service account, the set of allowable SCCs includes any constraints accessible to the service account.
+====
+
```
Comment (stlaz) [inline modules/security-context-constraints-about.adoc:15]: This information is already present in https://docs.openshift.com/container-platform/4.10/authentication/managing-security-context-constraints.html#admission_configuring-internal-oauth. Possibly outline it there.

**Review by stlaz — CHANGES_REQUESTED** [review]

File: modules/security-context-constraints-about.adoc (hunk 1 above)
Comment (cardil) [reply modules/security-context-constraints-about.adoc:15]: I do not find this explanation of this discrepancy on the page you've linked.

This is confusing, so I'd say having it here in plain text like this is beneficial, as people run pods using `kubeadmin` account. Of course, we could link to https://docs.openshift.com/container-platform/4.10/authentication/managing-security-context-constraints.html for more details.

Comment (xenolinux) [reply modules/security-context-constraints-about.adoc:15]: @stlaz @cardil so should we keep the note as is?

Comment (stlaz) [reply modules/security-context-constraints-about.adoc:15]: Quoting the page I linked earlier
> The set of SCCs that admission uses to authorize a pod are determined by the user identity and groups that the user belongs to. Additionally, if the pod specifies a service account, the set of allowable SCCs includes any constraints accessible to the service account.

I agree that it is confusing and does not lead directly to the conclusions made by @xenolinux here.

I would propose to move this note somewhere to the explanation quoted earlier.

However, I would also ask to word the new note a bit better. Specifically:
> There is a difference in permissions experienced by containers

I don't understand this sentence. Containers don't "experience" permissions.

> higher level type

Deployment is not a type. You can call these "Pod controllers" if you would.

> use secured `ServiceAccount` access

I don't understand this part, how is this access secured?

> but creating the pod uses `kubectl` or `oc` user access

You can create the pod directly even without kubectl or oc. You can create the pod using a controller with kubectl or oc. I propose to use "create directly".

> which may exceed safe {product-title} standards.

I would leave this out, it's misleading

> The set of SCCs that admission uses to authorize a pod are determined by the user identity and groups that the user belongs to. Additionally, if the pod specifies a service account, the set of allowable SCCs includes any constraints accessible to the service account.

This is a copy-paste of the paragraph I was pointing to originally, by moving this note to my proposed spot you can remove this part.

**Comment by wangke19** [comment]
My suggestion is to add this explanation to the known issues list. For the common user, they don't need to focus on that.

> There is a difference in permissions experienced by containers 
SCC is to control permissions for pods, not containers

I am not the right person of SCC for QA review. Transfer to @xingxingxia.

**Comment by xingxingxia** [comment]
@gkarager could you help review?

**Comment by gkarager** [comment]
I agree with @stlaz, this information is repetitive in the same page
> The set of SCCs that admission uses to authorize a pod are determined by the user identity and groups..

File: modules/security-context-constraints-about.adoc (hunk 1 above)
Comment (xenolinux) [reply modules/security-context-constraints-about.adoc]: Addressed

**Comment by xenolinux** [comment]
> I agree with @stlaz, this information is repetitive in the same page
> 
> > The set of SCCs that admission uses to authorize a pod are determined by the user identity and groups..

Addressed

**Comment by xenolinux** [comment]
@stlaz Would you please take a look at this PR?

File: modules/security-context-constraints-about.adoc [hunk 2]
```diff
@@ -319,6 +319,11 @@ user identity and groups that the user belongs to. Additionally, if the pod
 specifies a service account, the set of allowable SCCs includes any constraints
 accessible to the service account.
 
+[IMPORTANT]
+====
+There is a difference in permissions applied by containers, when you create pods directly by `kubectl` or `oc` and when pods are being created by some pod controllers like deployment or job. The pods created by higher level resources use `ServiceAccount` access.
```
Comment (stlaz) [inline modules/security-context-constraints-about.adoc:324]:
```suggestion
When creating pods directly, SCCs admission considers SCC permissions of both the caller and the Service Account that will run the pod.
When a pod is being created by a pod controller (such as a deployment or a job), only Service Account SCC permissions are considered.
```

**Review by stlaz — CHANGES_REQUESTED** [review]

File: modules/security-context-constraints-about.adoc (hunk 2 above)
Comment (xenolinux) [reply modules/security-context-constraints-about.adoc]: Addresed

**Comment by gkarager** [comment]
lgtm from QE side

**Review by skrthomas — COMMENTED** [review]
Lgtm!

File: modules/security-context-constraints-about.adoc [hunk 3]
```diff
@@ -319,6 +319,12 @@ user identity and groups that the user belongs to. Additionally, if the pod
 specifies a service account, the set of allowable SCCs includes any constraints
 accessible to the service account.
 
+[IMPORTANT]
+====
+When creating pods directly, SCCs admission considers SCC permissions of both the caller and the Service Account that runs the pod.
+When a pod is created by a pod controller such as a deployment or a job, only Service Account SCC permissions are considered.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/security-context-constraints-about.adoc]: Addressed

**Review by kelbrown20 — COMMENTED** [review]
/lgtm

**Pull request merged**